### PR TITLE
Centralize ReceiverType conversions

### PR DESCRIPTION
**Pull request by sellout**
Small refactor in preparation for other changes.

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -3242,19 +3284,15 @@ UniValue z_getaddressforaccount(const UniValue& params, bool fHelp)
...
-                throw JSONRPCError(RPC_INVALID_PARAMETER, "receiver type arguments must be \"p2pkh\", \"sapling\", or \"orchard\"");
-            }
-        }
+        receiverTypes = ReceiverTypesFromJSON(params[1].get_array())
+            .map_error([](const std::set<std::string>& invalidReceivers) {
+                throw JSONRPCError(
+                        RPC_INVALID_PARAMETER,
+                        strprintf(
+                            "\"%s\" is an invalid receiver type. Arguments must be \"p2pkh\", \"sapling\", or \"orchard\"",
+                            *invalidReceivers.begin()));
```
Comment (nuttycom) [inline src/wallet/rpcwallet.cpp:3293]:
```suggestion
                        strprintf(
                            "\"%s\" %s. Arguments must be \"p2pkh\", \"sapling\", or \"orchard\"",
                            invalidReceivers.size() > 1 ? "are invalid receiver types" : "is an invalid receiver type",
                            *invalidReceivers.begin()));
```

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -3333,6 +3371,10 @@ UniValue z_getaddressforaccount(const UniValue& params, bool fHelp)
                     strErr = tfm::format(
                         "Error: one or more of the requested receiver types does not have a corresponding spending key in this account.");
                     break;
+                case UnifiedAddressGenerationError::ReceiverTypeNotSupported:
+                    strErr = tfm::format(
+                        "Error: P2SH addresses can not be created by zcashd.");
```
Comment (nuttycom) [inline src/wallet/rpcwallet.cpp:3376]:
```suggestion
                        "Error: P2SH addresses can not be created using this RPC method.");
```

**Review by nuttycom — APPROVED** [review]
utACK modulo suggestions

File: src/wallet/rpcwallet.cpp (hunk 2 above)
Comment (sellout) [reply src/wallet/rpcwallet.cpp:3376]: Are there actually ways to create P2SH addresses in zcashd?

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (sellout) [reply src/wallet/rpcwallet.cpp:3293]: Yeah, I can do this, but it needs a bigger change than this suggestion – I was just lazy and avoiding iterating over the set.

File: src/wallet/rpcwallet.cpp (hunk 2 above)
Comment (nuttycom) [reply src/wallet/rpcwallet.cpp:3376]: Yes, using the `addmultisigaddress` RPC API method.

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (sellout) [reply src/wallet/rpcwallet.cpp]: I overkilled this for you.

**Review by nuttycom — APPROVED** [review]
utACK

**Pull request merged**